guac_core/src/api.rs

```diff
@@ -22,49 +21,23 @@ impl<T> NetworkRequest<T> {
     }
 }
 
-pub fn update(update: NetworkRequest<UpdateTx>) -> impl Future<Item = UpdateTx, Error = Error> {
-    STORAGE
-        .get_channel(update.from_addr.clone())
-        .and_then(move |mut channel_manager| {
-            channel_manager.received_payment(&update.data)?;
-            channel_manager.create_payment()
-        })
+pub fn update(update: NetworkRequest<UpdateTx>) -> Box<Future<Item = UpdateTx, Error = Error>> {
```

Comment:
Why was all this code removed? This stuff seems like something from the original codebase that we can actiually keep

Reply:
@jtremback it was removed because there were compilation errors that were affecting implementation of the flow. The logic behind this code is good and should be reused, but in case of `PaymentManager` trait implementation it could look slightly different. Its still in the history though.